tests/python/openassetio/managerAPI/test_hostsession.py

```diff
@@ -40,14 +41,39 @@ def mock_logger():
 
 
 @pytest.fixture
-def host_session(mock_host, mock_logger):
-    return HostSession(mock_host, mock_logger)
+def host_session(a_host, mock_logger):
+    return HostSession(a_host, mock_logger)
 
 
-class TestHostSession():
+class Test_HostSession_init:
+    def test_when_invalid_host_then_raises_TypeError(self, mock_logger):
+        with pytest.raises(TypeError) as err:
+            HostSession(object(), mock_logger)
+
+        assert str(err.value).startswith("__init__(): incompatible constructor arguments")
+
+
+class Test_HostSession_host:
+    def test_returns_copy_of_host(self, host_session, a_host, mock_host_interface):
+        expected_identifier = "some identifier"
+        mock_host_interface.mock.identifier.return_value = expected_identifier
+
+        actual_host = host_session.host()
 
-    def test_construction(self, host_session, mock_host):
-        assert host_session.host() is mock_host
+        # TODO(DF): the following assertion fails, since the C++
+        #   HostSession takes a Host by-value and is non-polymorphic,
+        #   so is no longer the Python subclass instance when queried.
+        #   This will be fixed when the pure-Python Host is removed, at
+        #   which point we should make Host a `final` class in both C++
+        #   and pybind.
+        # assert isinstance(actual_host, Host)
+        assert isinstance(actual_host, _openassetio.managerAPI.Host)
+
+        assert actual_host is not a_host
+        assert actual_host.identifier() == expected_identifier
```

Comment:
[fix] Im not sure this is a helpful test - as the identifiers matching doesn't guarantee that the host returns is holding the same interface instance as the supplied host - which is what we really _must_ ensure.

Reply:
I guess this maybe raises the question that do we provide an equality operator for `Host` that compares the interface ptr?

Reply:
The Python `Host` class has `_interface()`, which we could migrate and check that for equality?

Reply:
Yeah - that was def one of the things on the list to re-evaluate at this point.

Reply:
Much simplified by the move to `SharedPtr` in 3ca11608